var/spack/repos/builtin/packages/m4/package.py

```diff
@@ -51,6 +51,9 @@ def configure_args(self):
         if spec.satisfies('%clang') and not spec.satisfies('platform=darwin'):
             args.append('CFLAGS=-rtlib=compiler-rt')
 
+        if spec.satisfies('%intel'):
+            args.append('CFLAGS=-no-gcc')
```

Comment:
Would that overwrite user defined `CFLAGS`?

Reply:
Yes, it feels like a hack. The same with the previous `clang` part that is already in the `develop` branch.

Reply:
I tried `./configure CFLAGS="-O1 -g" CFLAGS="-O2"` on a different package and only `-O2` is picked up for building.

I think we want to use `cflags.append(-no-gcc)` and the same for the above flag.

Reply:
Fixed in 4cb1c2b1d7540a75d7b17312d2732087fbf94587.

Reply:
@junghans I don't see how the change you requested can work. I did try it, but spack doesn't pick up this change and says the package was already built. But I think the `cflags` is just a local variable that will not do anything.

Reply:
I will be off my computer. If you figure it out, feel free to push into this PR, you should have the rights.